Backend/controller/FlightController.js

```diff
@@ -48,61 +48,91 @@ router.delete("/:id", async (req, res) => {
   res.json(numberOfDestroyedRows);
 });
 
-// @desc Saves a new flight to the database
+// @desc Performs a check on the G-Record of a provided IGC-File and if valid persists the IGC-File.
 // @route POST /flight/
 
 router.post("/", async (req, res) => {
-  const newFlight = req.body;
-  newFlight.flightStatus = "In Bearbeitung";
+  const igc = req.body.igc;
+  const userId = req.body.userId;
 
-  if (areRequiredFlightParamsInvalid(newFlight)) {
-    res.sendStatus(BAD_REQUEST);
+  try {
+    checkParamsForIgc(igc);
+  } catch (error) {
+    res.status(BAD_REQUEST).send(error);
     return;
   }
 
-  const igcFile = newFlight.igc;
-  delete newFlight.igc;
-
-  //TODO Handle result; What happens if service not reachable?
-  igcValidator.execute(igcFile);
-
-  service.save(newFlight).then((result) => {
-    writeIgcFileToDrive(result.id, igcFile)
-      .then(() => {
-        service.startResultCalculation(result);
-        //TODO Replace the later statement
-        //Return a hard coded flight id and takeoff & landing for development. Should be the data in future.
-        res.status(OK).send({
-          flightId: result.id,
-          takeoff: "Bremm",
-          landing: "Zeltingen-Rachtig",
-        });
-      })
-      .catch((error) => {
-        res.status(INTERNAL_SERVER_ERROR).send(error.message);
+  igcValidator
+    .execute(igc)
+    .then((result) => {
+      if (result == "FAILED") {
+        // res.status(BAD_REQUEST).send("Invalid G-Record");
+        // return;
+        // TODO Current example is invalid! Repair it!
+      }
+      service
+        .create({
+          userId: userId,
+        })
+        .then((flight) =>
+          persistIgcFile(flight.id, igc).then((igcUrl) => {
+            flight.igcUrl = igcUrl;
+            service.update(flight).then((flight) => {
+              res.json({
+                flightId: flight.id,
+                takeoff: "Bremm",
+                landing: "Zeltingen-Rachtig",
+              });
+            });
+          })
+        );
+    })
+    .catch((error) => {
+      //Couldn't execute request to FAI API
+      res.status(BAD_REQUEST).send(error);
+    });
+});
+
+// @desc Adds futher data to a existing flight and starts the igc calculation if no igc result are present.
+// @route PUT /flight/:id
+
+router.put("/:id", async (req, res) => {
+  const report = req.body.report;
+  const glider = req.body.glider;
```

Comment:
Glider is a UUID or a string?
Maybe needs to be checked to be a glider of the user.

Reply:
For now it's just a string. Glider has currently no effect on scoring.
Should be a object with brand, name and class in the future.